### PR TITLE
bug(celery): Refactor mailer and celery function

### PR DESCRIPTION
**Pull request by Zenjjim**
# PR NAME
## Description
Update save/create method for better flow.
Update mailer functions

Comments/issues:
Would love to write tests for this, however I would need some help understanding how
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] PR name is descriptive
- [x] The PR includes the link to the relevant Trello task
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [ ] `python manage.py runserver` was run locally witout errors

## Pull request type

Please check the type of change your PR introduces:
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: app/content/signals.py [hunk 1]
```diff
@@ -2,42 +2,46 @@
...
+                    "evaluate_link": event.evaluate_link,
+                },
+            )
+        if should_send_incoming_sign_off_deadline_reminder(event):
+            event.sign_off_deadline_schedular_id = event_sign_off_deadline_schedular.apply_async(
+                eta=(midday(event.sign_off_deadline - timedelta(days=1))),
+                kwargs={"pk": event.pk, "title": event.title},
+            )
+    except Exception as e:
+        print(e)
```
Comment (Eirsteir) [inline app/content/signals.py:37]: I think we should start using `logger.info` instead to prepare for actual logging. For now the logger will still log to the console

File: app/content/signals.py [hunk 2]
```diff
@@ -2,42 +2,46 @@
...
 
 
 def should_send_incoming_end_date_reminder(event):
-    return event.evaluate_link and event.end_date < today() and not event.closed
+    event_has_ended = event.end_date < today()
```
Comment (Eirsteir) [inline app/content/signals.py:41]: This and the one in the method below are perfect candidates for properties on the Event model 😄

File: app/content/signals.py [hunk 3]
```diff
@@ -2,42 +2,46 @@
...
+                eta=(midday(event.end_date + timedelta(days=1))),
+                kwargs={
+                    "pk": event.pk,
+                    "title": event.title,
+                    "start_date": datetime_format(event.start_date),
+                    "evaluate_link": event.evaluate_link,
+                },
+            )
+        if should_send_incoming_sign_off_deadline_reminder(event):
+            event.sign_off_deadline_schedular_id = event_sign_off_deadline_schedular.apply_async(
```
Comment (Eirsteir) [inline app/content/signals.py:32]: I am just wondering about this assignment and the one above. Would we not have to save the event after both?

**Review by Eirsteir — CHANGES_REQUESTED** [review]
Good work celery master 🥕

File: app/content/models/registration.py [hunk 4]
```diff
@@ -38,59 +35,24 @@ def __str__(self):
     def save(self, *args, **kwargs):
         """ Determines whether the object is being created or updated and acts accordingly """
         if not self.registration_id:
-            return self.create(*args, **kwargs)
+            self = self.create()
```
[the remark this answers is not in the document]
Comment (Zenjjim) [reply app/content/models/registration.py:38]: Agreed, this is unnecessary

File: app/content/signals.py (hunk 1 above)
Comment (Zenjjim) [reply app/content/signals.py:37]: Good idea. I mainly use if for debugging but starting to log is a good idea

File: app/content/signals.py (hunk 2 above)
Comment (Zenjjim) [reply app/content/signals.py:41]: Good idea

File: app/content/signals.py (hunk 3 above)
Comment (Zenjjim) [reply app/content/signals.py:32]: you are right. I totally forgot :D

**Comment by Zenjjim** [comment]
1 runde QA kommentarer fikset :D

**Review by Eirsteir — CHANGES_REQUESTED** [review]
Super! I have only one last comment. Please take a look at it before approval😄

**Review by Eirsteir — APPROVED** [review]
Fantastic job moving celery and mailing logic out of the registration model! 🎉

**Pull request merged**